src/io/flutter/FlutterUtils.java

```diff
@@ -58,4 +61,15 @@ public static VirtualFile getRealVirtualFile(@Nullable PsiFile psiFile) {
   public static void openFlutterSettings(@Nullable Project project) {
     ShowSettingsUtilImpl.showSettingsDialog(project, FlutterConstants.FLUTTER_SETTINGS_PAGE_ID, "");
   }
+
+  /**
+   * Checks whether a given string is a valid Dart identifier.
+   *
+   * See: https://www.dartlang.org/guides/language/spec
+   * @param id the string to check
+   * @return true if a valid identifer, false otherwise.
+   */
+  public static boolean isValidDartdentifier(@NotNull String id) {
```

Comment:
In the past we had issues w/ users choosing project names that were the same as common pub package names (esp. the names of flutter dependencies), and running into issues w/ the first `pub get`. For `flutter create`, we proscribe the common package names: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/create.dart#L248.